src/simpleDetFunc.c

```diff
@@ -376,21 +378,21 @@ int simpleDetFunctionCalculation(int correntropyWinSize, int interval,
 	 * function that computes correntropy. Add correntropyWinSize zeros 
 	 * will be sufficient.
 	 */
-	bufferLength = dataLength + correntropyWinSize;
+	bufferLength = (numWindows-1)*interval + 2*correntropyWinSize + 2;
 	printf("bufferLength: %d\n",bufferLength);
 	buffer = malloc(sizeof(float)*bufferLength);
 	for (i = dataLength; i<bufferLength;i++){
 		buffer[i] = 0;
 	}
 
-	
 	pooledSummaryMatrix = malloc(sizeof(float)*numWindows);
 	for (i=0;i<numWindows;i++){
 		pooledSummaryMatrix[i] = 0;
 	}
 	sigmas = malloc(sizeof(float)*numWindows);
 
-	centralFreq = centralFreqMapper(numChannels, minFreq, maxFreq);
+	centralFreq = malloc(sizeof(float)*numChannels);
```

Comment:
Whenever there is a malloc call you should always check if NULL is returned, and return an error accordingly. You do it when calling this same function in filterbank.c in this commit.